js/client/modules/@arangodb/testutils/testrunner.js

```diff
@@ -181,6 +182,55 @@ let viewsTest = {
   }
 };
 
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief checks that no new analyzers were left on the SUT. 
+// //////////////////////////////////////////////////////////////////////////////
+let aralyzersTest = {
+  name: 'analyzers',
+  setUp: function(obj, te) {
+    analyzers.toArray().forEach(oneAnalyzer => {
+      obj.analyzersBefore.push(oneAnalyzer.name());
+    });
+    return true;
+  },
+  runCheck: function(obj, te) {
+    let leftover = [];
+    let foundAnalyzers = [];
+    try {
+      analyzers.toArray().forEach(oneAnalyzer => {
+        let name = oneAnalyzer.name();
+        let found = obj.analyzersBefore.find(oneAnalyzer => oneAnalyzer === name);
+        if (found === name) {
+          foundAnalyzers.push(name);
+        } else {
+          leftover.push(name);
+        }
+      });
+    } catch (x) {
+      obj.results[obj.translateResult(te)] = {
+        status: false,
+        message: 'failed to fetch the currently available analyzers: ' + x.message + '. Original test status: ' + JSON.stringify(obj.results[obj.translateResult(te)])
+      };
+      return false;
+    }
+    if (leftover.length !== 0) {
+      obj.results[obj.translateResult(te)] = {
+        status: false,
+        message: 'Cleanup missing - test left over analyzer:' + leftover + '. Original test status: ' + JSON.stringify(obj.results[obj.translateResult(te)])
```

Comment:
Will this message work as expected if `leftover` would be an array with more than 1 element?

Reply:
yes, it creates a coma separated list